graphql/http_handler.go

```diff
@@ -23,7 +30,33 @@ func Handler(apiURL string) func(w http.ResponseWriter, r *http.Request) {
 	// Send OTEL traces for each request.
 	// Only create spans for resolved fields.
 	srv.Use(otelgqlgen.Middleware(
-		otelgqlgen.WithCreateSpanFromFields(func(fieldCtx *graphql.FieldContext) bool { return fieldCtx.IsResolver }),
+		otelgqlgen.WithCreateSpanFromFields(func(fieldCtx *graphql.FieldContext) bool {
+			return fieldCtx.IsMethod
+		}),
+		otelgqlgen.WithRequestVariablesAttributesBuilder(
+			otelgqlgen.RequestVariablesBuilderFunc(func(requestVariables map[string]interface{}) []attribute.KeyValue {
+				// Deep clone the request variables to avoid modifying the original map.
+				redactedRequestVariables := map[string]interface{}{}
+				registeredTypes := []interface{}{
+					map[string]interface{}{},
+				}
+				// We need to deep copy the request variables to avoid modifying the original map. Since it is passed on to the next middleware.
+				if err := util.DeepCopy(requestVariables, &redactedRequestVariables, registeredTypes); err != nil {
+					grip.Error(message.WrapError(err, message.Fields{
+						"message": "failed to deep copy request variables",
+					}))
+					return nil
+				}
+				RedactFieldsInMap(redactedRequestVariables, redactedFields)
+				variables := make([]attribute.KeyValue, 0, len(redactedRequestVariables))
+
+				for k, v := range redactedRequestVariables {
```

Comment:
Maybve this could just call `otelgqlgen.RequestVariables` with the new map?

Reply:
Good call!